frontend/packages/kubevirt-plugin/integration-tests/tests/models/baseVMBuilder.ts

```diff
@@ -0,0 +1,145 @@
+import * as _ from 'lodash';
+import { OperatingSystem, Workload, OSIDLookup } from '../utils/constants/wizard';
+import { FlavorConfig, ProvisionSource, Disk, Network, CloudInitConfig } from '../types/types';
+import { BaseVMBuilderData } from '../types/vm';
+import { K8sKind } from '@console/internal/module/k8s';
+import { getRandStr } from '../utils/utils';
+
+export abstract class BaseVMBuilder<T extends BaseVMBuilderData> {
+  protected kind: K8sKind;
+
+  protected data: T;
+
+  constructor(kind: K8sKind, builder?: BaseVMBuilder<T>) {
+    this.kind = kind;
+    this.data = builder
+      ? builder.getData()
+      : ({ networks: [], disks: [], storageClassAttributes: {} } as any);
+  }
+
+  public setName(name: string) {
+    this.data.name = name;
+    return this;
+  }
+
+  public generateName(id?: string) {
+    const finalId = id || getRandStr(5);
+    const osId = OSIDLookup[this.data.os];
+    this.data.name = [
+      this.kind.abbr.toLowerCase(),
+      this.data.namespace,
+      finalId,
+      this.data.flavor?.flavor,
+      osId,
+    ]
+      .filter((a) => a)
+      .map((a) => a.toLowerCase().replace(/[^-a-zA-Z0-9]/g, ''))
+      .join('-');
+    return this;
+  }
+
+  public setDescription(description: string) {
+    this.data.description = description;
+    return this;
+  }
+
+  public setNamespace(namespace: string) {
+    this.data.namespace = namespace;
+    return this;
+  }
+
+  public setTemplate(template: string) {
+    this.data.template = template;
+    return this;
+  }
+
+  public setFlavor(flavor: FlavorConfig) {
+    this.data.flavor = flavor;
+    return this;
+  }
+
+  public setWorkload(workload: Workload) {
+    this.data.workload = workload;
+    return this;
+  }
+
+  public setOS(os: OperatingSystem) {
+    this.data.os = os;
+    return this;
+  }
+
+  public setProvisionSource(provisionSource: ProvisionSource) {
+    this.data.provisionSource = provisionSource;
+    return this;
+  }
+
+  public setNetworks(networks: Network[]) {
+    this.data.networks = networks;
+    return this;
+  }
+
+  public setDisks(disks: Disk[]) {
+    this.data.disks = disks;
+    return this;
+  }
+
+  public setStorageAttributes(storageAttributes: { [k: string]: string }) {
+    this.data.storageClassAttributes = storageAttributes;
```

Comment:
are these used anywhere? Or expected to be used? 

Also if yes, then we should define a type for them so it is referencable throughout the code (not generic map of strings)

Reply:
Currently, we're not using this at all and define the attrubutes in a Disk object. I'll remove this from this PR for now, there is one use case at this moment where this could be used, which is `ID(CNV-4096) Creates DV with user-selected accessMode/volumeMode`

Reply:
+1, let's add it when we need it

Reply:
removed for now